csi/controller.go

```diff
@@ -602,11 +602,14 @@ func resolveSharedSpec(spec *api.VolumeSpec, req *csi.CreateVolumeRequest) (*api
 	}
 
 	// Handle legacy support for shared
-	if spec.Shared && !spec.Sharedv4 {
-		spec.Shared = shared
-	} else {
-		// assume sharedv4 by default
-		spec.Sharedv4 = shared
+	// shared or sharedv4 parameter doesn't apply to pure backends so don't set them
+	if !spec.IsPureVolume() {
```

Comment:
Let's move this to the beginning of `resolveSharedSpec`:

```
if spec.IsPureVolume() {
  return spec, nil
}
```

Will make this case easier to read.